pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2114,12 +2127,20 @@ func (a *AWSCloud) getInstanceById(instanceID string) (*ec2.Instance, error) {
 		return nil, err
 	}
 	if len(instances) == 0 {
```

Comment:
I suggest moving this test into getInstanceByID.  Then we avoid the problem where we return an error from ListRoutes if we have no routes.